superset-frontend/src/SqlLab/components/SqlEditor/index.jsx

```diff
@@ -362,176 +349,163 @@ class SqlEditor extends React.PureComponent {
     const { aceEditorHeight, southPaneHeight } =
-      this.getAceEditorAndSouthPaneHeights(
-        this.state.height,
-        this.state.northPercent,
-        this.state.southPercent,
-      );
+      getAceEditorAndSouthPaneHeights(height, northPercent, southPercent);
     return (
       <Split
         expandToMin
         className="queryPane"
-        sizes={[this.state.northPercent, this.state.southPercent]}
-        elementStyle={this.elementStyle}
+        sizes={[northPercent, southPercent]}
+        elementStyle={elementStyle}
         minSize={200}
         direction="vertical"
         gutterSize={SQL_EDITOR_GUTTER_HEIGHT}
-        onDragStart={this.onResizeStart}
-        onDragEnd={this.onResizeEnd}
+        onDragStart={onResizeStart}
+        onDragEnd={onResizeEnd}
       >
-        <div ref={this.northPaneRef} className="north-pane">
+        <div ref={northPaneRef} className="north-pane">
           <AceEditorWrapper
-            actions={this.props.actions}
-            autocomplete={this.state.autocompleteEnabled}
-            onBlur={this.setQueryEditorSql}
-            onChange={this.onSqlChanged}
-            queryEditor={this.props.queryEditor}
-            database={this.props.database}
-            extendedTables={this.props.tables}
+            actions={actions}
+            autocomplete={autocompleteEnabled}
+            onBlur={setQueryEditorAndSaveSql}
+            onChange={onSqlChanged}
+            queryEditor={queryEditor}
+            database={database}
+            extendedTables={tables}
             height={`${aceEditorHeight}px`}
             hotkeys={hotkeys}
           />
-          {this.renderEditorBottomBar(hotkeys)}
+          {renderEditorBottomBar(hotkeys)}
         </div>
         <ConnectedSouthPane
-          editorQueries={this.props.editorQueries}
-          latestQueryId={this.props.latestQuery && this.props.latestQuery.id}
-          dataPreviewQueries={this.props.dataPreviewQueries}
-          actions={this.props.actions}
+          editorQueries={editorQueries}
+          latestQueryId={latestQuery && latestQuery.id}
+          dataPreviewQueries={dataPreviewQueries}
+          actions={actions}
           height={southPaneHeight}
-          displayLimit={this.props.displayLimit}
-          defaultQueryLimit={this.props.defaultQueryLimit}
+          displayLimit={displayLimit}
+          defaultQueryLimit={defaultQueryLimit}
         />
       </Split>
     );
-  }
+  };
 
-  renderDropdown() {
-    const qe = this.props.queryEditor;
-    const successful = this.props.latestQuery?.state === 'success';
+  const renderDropdown = () => {
+    const qe = queryEditor;
+    const successful = latestQuery?.state === 'success';
     const scheduleToolTip = successful
       ? t('Schedule the query periodically')
       : t('You must run the query successfully first');
     return (
-      <Menu onClick={this.handleMenuClick} style={{ width: 176 }}>
+      <Menu style={{ width: 176 }}>
```

Comment:
```suggestion
      <Menu css={{ width: theme.gridUnit * 44 }}>
```
Unit values should stay within the Superset theme, also use `css` instead of `style`